certbot/tests/main_test.py

```diff
@@ -960,15 +958,12 @@ def test_certonly_new_request_success(self, mock_get_utility, mock_notAfter, moc
         self.assertTrue(key_path in cert_msg)
         self.assertTrue(
             'donate' in mock_get_utility().add_message.call_args[0][0])
-        self.assertTrue(mock_sub.called)
```

Comment:
Here and for the other comments about tests removed: I think you made the review before the last commits. In the final diff, these lines are not modified.

Reply:
Yeah that's my point. You previously had added tests that `handle_subscription` was or was not called in `test_certonly_new_request_success`, `test_certonly_new_request_failure`, and `test_certonly_csr` but you removed them.

I was curious why you removed them because they seemed like good tests to me.

Reply:
OK, I understand. It is the side effect of the reverts I did of these commits, when I changed the overall logic. I wanted to avoid to add unnecessary mocks on tests that were working again without them once on the new logic.

I will add them back, if relevant.

Reply:
Sounds good. Let me know when this is ready for another review.